### PR TITLE
feat: refactoring calculator

### PR DESCRIPTION
**Pull request by eriklimakc**

### CODE REVIEW

**Review by dansiegel — CHANGES_REQUESTED** [review]

File: reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs [hunk 1]
```diff
@@ -45,137 +62,148 @@ private Calculator Input(Calculator calculator, string key)
...
+                        break;
+                    case "×":
+                        result = calculator.Number1!.Value * number2!.Value;
+                        break;
+                    case "+":
+                        result = calculator.Number1!.Value + number2!.Value;
+                        break;
+                    case "-":
+                        result = calculator.Number1!.Value - number2!.Value;
+                        break;
```
[the remark this answers is not in the document]
Comment (eriklimakc) [reply reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs]: Fix in my next commit.

File: reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs [hunk 2]
```diff
@@ -45,137 +62,148 @@ private Calculator Input(Calculator calculator, string key)
...
+                    case "×":
+                        result = calculator.Number1!.Value * (number2!.Value / 100) * calculator.Number1!.Value;
+                        break;
+                    case "+":
+                        result = calculator.Number1!.Value + (number2!.Value / 100) * calculator.Number1!.Value;
+                        break;
+                    case "-":
+                        result = calculator.Number1!.Value - (number2!.Value / 100) * calculator.Number1!.Value;
+                        break;
```
[the remark this answers is not in the document]
Comment (eriklimakc) [reply reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs]: Fix in my next commit.

**Comment by nickrandolph** [comment]
@eriklimakc can this be merged into the 4-7 branch?

**Comment by eriklimakc** [comment]
@nickrandolph Not yet, I'm working on a few things Dan and I planned. Should be ready by the end of the day, though.

**Comment by nickrandolph** [comment]
No probs. I pushed this and the other calculator PR back to being in draft until they're ready for review and merge

File: reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs [hunk 3]
```diff
@@ -1,183 +1,238 @@
...
+        { KeyInput.One, "1" },
+        { KeyInput.Two, "2" },
+        { KeyInput.Three, "3" },
+        { KeyInput.Four, "4" },
+        { KeyInput.Five, "5" },
+        { KeyInput.Six, "6" },
+        { KeyInput.Seven, "7" },
+        { KeyInput.Eight, "8" },
+        { KeyInput.Nine, "9" }
+    };
```
Comment (dansiegel) [inline reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs:31]: we don't need this... you have an enum... the conditional logic should just be checking the enum value

File: reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs [hunk 4]
```diff
@@ -1,183 +1,238 @@
...
-        private bool HasNumber2 => Number2 != null;
-
-        public string Output => $"{(Result != null ? Result.Value : HasNumber ? Number : "0")}";
-        public string Equation => $"{Number1} {Operator} {Number2}{(IsNumber2Percentage ? "%" : string.Empty)}{(Result != null ? " =" : string.Empty)}";
-
-        public Calculator Input(string key)
-            => Input(this, key);
-
-        private Calculator Input(Calculator calculator, string key)
+        if (KeyTranslator.TryGetValue(key, out var keyValue))
```
Comment (dansiegel) [inline reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs:52]: we already have a set value with the enum for the keys we shouldn't be trying to get a symbol here

File: reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs [hunk 5]
```diff
@@ -1,183 +1,238 @@
...
+        {
+            throw new InvalidOperationException("Inputed key is not valid.");
+        }
+    }
+
+    private Calculator RestartOrClear(Calculator calculator, string key)
+    {
+        if (calculator.Result != null)
+        {
+            if (key == "÷" || key == "×" || key == "+" || key == "-")
```
Comment (dansiegel) [inline reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs:80]: use the enum

File: reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs [hunk 6]
```diff
@@ -1,183 +1,238 @@
...
+    Zero,
+    One,
+    Two,
+    Three,
+    Four,
+    Five,
+    Six,
+    Seven,
+    Eight,
+    Nine
```
Comment (dansiegel) [inline reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs:237]: save a headache and put these at the beginning of the enum like:
```cs
Zero = 0,
One = 1,
Two = 2
```
This way when it is a numeric value you can just cast to an int and append it.

**Review by dansiegel — CHANGES_REQUESTED** [review]

File: reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs (hunk 4 above)
Comment (eriklimakc) [reply reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs]: Changes in next commit

File: reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs (hunk 5 above)
Comment (eriklimakc) [reply reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs]: Changes in next commit

File: reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs (hunk 6 above)
Comment (eriklimakc) [reply reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs]: Changes in next commit

File: reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs (hunk 3 above)
Comment (eriklimakc) [reply reference/SimpleCalculator/SimpleCalculator.Shared/Business/Calculator.cs]: Changes in next commit

**Review by dansiegel — APPROVED** [review]

**Pull request merged**